File: src/main/kotlin/net/thexcoders/aladin_bot_backend/converters/CategoryConverter.java
package net.thexcoders.aladin_bot_backend.converters;

public class CategoryConverter {

    public static final int UNKNOWN_CAT = -1;
    public static final int GREETINGS = 0;
    public static final int CONV_END = 1;
    public static final int EMERGENCY = 2;
    public static final int CREATE_DATA = 3;
    public static final int TIME_DATA = 4;
    public static final int BEST_PLACE = 5;
    public static final int BEST_FOOD = 6;

    /**
     * Converting the Category code to an Integer.
     *
     * @param lang String representing the category
     * @return Integer Code of the category
     */


    public static int categoryConverter(String lang) {
        switch (lang) {
            case "greeting":
                return GREETINGS;
            case "conv-end":
                return CONV_END;
            case "emergency":
                return EMERGENCY;
            case "create-data":
                return CREATE_DATA;
            case "time-data":
                return TIME_DATA;
            case "best-place":
                return BEST_PLACE;
            case "best-food":
                return BEST_FOOD;
            default:
                return UNKNOWN_CAT;
        }
    }
}

File: src/main/kotlin/net/thexcoders/aladin_bot_backend/converters/CategoryConverter.kt
package net.thexcoders.aladin_bot_backend.converters

const val UNKNOWN_CAT = -1;
const val GREETINGS = 0;
const val CONV_END = 1;
const val EMERGENCY = 2;
const val CREATE_DATA = 3;
const val TIME_DATA = 4;
const val BEST_PLACE = 5;
const val BEST_FOOD = 6;

fun categoryConverter(lang: String): Int {
    return when (lang) {
        "greeting" -> GREETINGS
        "conv-end" -> CONV_END
        "emergency" -> EMERGENCY
        "create-data" -> CREATE_DATA
        "time-data" -> TIME_DATA
        "best-place" -> BEST_PLACE
        "best-food " -> BEST_FOOD
        else -> UNKNOWN_CAT
    }
}